segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentMetadata.java

```diff
@@ -184,6 +185,13 @@ public synchronized long getLength() {
         return new AttributesView();
     }
 
+    @Override
+    public synchronized Map<UUID, Long> getAttributes(BiPredicate<UUID, Long> filter) {
```

Comment:
Is it necessary to make it a precondition that the filter argument is not null?

Reply:
Sure. I can do that. But whether I throw NullPtrEx now or whether it gets thrown on the next line really makes no difference. We'd just be adding more code to provide no extra value.

Reply:
Ok.